### PR TITLE
DDS Condition base class

### PR DESCRIPTION
**Pull request by IkerLuengo**
- class Condition created on PIM (with it's only method get_trigger_value)
- PSM mapping to PIM implementation

Unit tests will be added when derived GuardCondition is implemented, as this derived class implements the set_trigger_value() method and will provide a much richer testing

This must be merged after #1099

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1658/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1423/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1653/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1662/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1425/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1655/)

File: include/dds/core/cond/Condition.hpp [hunk 1]
```diff
@@ -94,12 +99,23 @@ class TCondition : public virtual Reference<DELEGATE>
      */
     template<typename Functor>
     OMG_DDS_API void handler(
-            Functor& func);
+            Functor& func)
+    {
+        //To implement
+        //    ISOCPP_REPORT_STACK_DDS_BEGIN(*this);
```
Comment (laura-eprosima) [inline include/dds/core/cond/Condition.hpp]:
```suggestion
```

File: include/dds/core/cond/Condition.hpp [hunk 2]
```diff
@@ -94,12 +99,23 @@ class TCondition : public virtual Reference<DELEGATE>
...
+    }
 
     /** @copydoc dds::core::cond::TCondition::handler(Functor& func) */
     template<typename Functor>
     OMG_DDS_API void handler(
-            const Functor& func);
+            const Functor& func)
+    {
+        //To implement
+        //    ISOCPP_REPORT_STACK_DDS_BEGIN(*this);
```
Comment (laura-eprosima) [inline include/dds/core/cond/Condition.hpp]:
```suggestion
```

File: include/dds/core/cond/Condition.hpp [hunk 3]
```diff
@@ -110,7 +126,12 @@ class TCondition : public virtual Reference<DELEGATE>
      * @return void
      * @throw  dds::core::Exception
      */
-    OMG_DDS_API void reset_handler();
+    OMG_DDS_API void reset_handler()
+    {
+        //To implement
+        //    ISOCPP_REPORT_STACK_DDS_BEGIN(*this);
```
Comment (laura-eprosima) [inline include/dds/core/cond/Condition.hpp]:
```suggestion
```

File: include/dds/core/cond/Condition.hpp [hunk 4]
```diff
@@ -121,7 +142,12 @@ class TCondition : public virtual Reference<DELEGATE>
      * @return void
      * @throw  dds::core::Exception
      */
-    OMG_DDS_API void dispatch();
+    OMG_DDS_API void dispatch()
+    {
+        //To implement
+        //    ISOCPP_REPORT_STACK_DDS_BEGIN(*this);
```
Comment (laura-eprosima) [inline include/dds/core/cond/Condition.hpp]:
```suggestion
```

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1683/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1444/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1675/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1704/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1465/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1696/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1708/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1469/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1700/)

**Pull request closed without merging**